lua/modules/diffusion/diffusion.cc

```diff
@@ -2,19 +2,162 @@
+
+  auto options_params = opensn::diffusion::CFEMSolver::OptionsBlock();
+  options_params.AssignParameters(params.GetParam("arg1"));
+
+  solver.SetOptions(options_params);
+
+  return opensn::ParameterBlock();
+}
+
+// CFEM diffusion solver
+
+opensn::ParameterBlock
+DFEMSetOptions(const opensn::InputParameters& params)
+{
+  const std::string fname = __FUNCTION__;
+
+  params.RequireParameter("arg0");
+  params.RequireParameter("arg1");
+
+  auto d_coef_function = CreateFunction("D_coef");
+  opensn::function_stack.push_back(d_coef_function);
+
+  auto q_ext_function = CreateFunction("Q_ext");
+  opensn::function_stack.push_back(q_ext_function);
+
+  auto sigma_a_function = CreateFunction("Sigma_a");
+  opensn::function_stack.push_back(sigma_a_function);
+
+  const size_t handle = params.GetParamValue<size_t>("arg0");
+  auto& solver =
+    opensn::GetStackItem<opensn::diffusion::DFEMSolver>(opensn::object_stack, handle, fname);
+  solver.SetDCoefFunction(d_coef_function);
+  solver.SetQExtFunction(q_ext_function);
+  solver.SetSigmaAFunction(sigma_a_function);
+
+  auto options_params = opensn::diffusion::DFEMSolver::OptionsBlock();
+  options_params.AssignParameters(params.GetParam("arg1"));
+
+  solver.SetOptions(options_params);
+
+  return opensn::ParameterBlock();
+}
+
+// FV diffusion solver
+
+opensn::ParameterBlock
+FVSetOptions(const opensn::InputParameters& params)
+{
+  const std::string fname = __FUNCTION__;
+
+  params.RequireParameter("arg0");
+  params.RequireParameter("arg1");
+
+  auto d_coef_function = CreateFunction("D_coef");
+  opensn::function_stack.push_back(d_coef_function);
+
+  auto q_ext_function = CreateFunction("Q_ext");
+  opensn::function_stack.push_back(q_ext_function);
+
+  auto sigma_a_function = CreateFunction("Sigma_a");
+  opensn::function_stack.push_back(sigma_a_function);
+
+  const size_t handle = params.GetParamValue<size_t>("arg0");
+  auto& solver =
+    opensn::GetStackItem<opensn::diffusion::FVSolver>(opensn::object_stack, handle, fname);
+  solver.SetDCoefFunction(d_coef_function);
+  solver.SetQExtFunction(q_ext_function);
+  solver.SetSigmaAFunction(sigma_a_function);
+
+  auto options_params = opensn::diffusion::FVSolver::OptionsBlock();
+  options_params.AssignParameters(params.GetParam("arg1"));
+
+  solver.SetOptions(options_params);
+
+  return opensn::ParameterBlock();
+}
+
+// CFEM MG diffusion solver
+
+opensn::ParameterBlock
+CFEMMGSetOptions(const opensn::InputParameters& params)
+{
+  const std::string fname = __FUNCTION__;
+
+  params.RequireParameter("arg0");
+  params.RequireParameter("arg1");
+
+  const size_t handle = params.GetParamValue<size_t>("arg0");
+  auto& solver =
+    opensn::GetStackItem<opensn::diffusion::MGSolver>(opensn::object_stack, handle, fname);
+
+  auto options_params = opensn::diffusion::MGSolver::OptionsBlock();
+  options_params.AssignParameters(params.GetParam("arg1"));
+
+  solver.SetOptions(options_params);
+
+  return opensn::ParameterBlock();
+}
```

Comment:
Is CFEM the only option in the MG solver at present? If so, that should be changed fixed later on.

Reply:
is the diffusion solvers are merged here, then CFEM and DFEM should be available options

Reply:
The MG solver only has CFEM option as far as I can tell..

Reply:
ok, so the description "Grouping diffusion solvers into a diffusion module" confused me. I see now that it is more a grouping of the files than a synthesis/merge of all of the flavors of diffusion solvers.

Reply:
In this sense, module is a collection of classes. Now, all solvers are in the diffusion namespace, both in the C++ and lua space.  The work that would introduce the common diffusion class, etc. would be called refactoring.